referendum/referendum.cpp

```diff
@@ -340,10 +346,10 @@ void referendum::clean(name account, name dac_id){
     votes_table votes(get_self(), dac_id.value);
     auto existing_vote_data = votes.find(account.value);
 
-    map<uint64_t, uint8_t> new_votes;
+    map<name, uint8_t> new_votes;
```

Comment:
I'm not sure I see the reason you have implemented the votes as a map when I think it might be simpler to use tables with separate fields. eg. `vote { autoincremeting_key, voter, referendum_id, vote_value}` Then a separate indices for `voter`, `referendumId` and `voter | referendumId` so you could search and filter as needed. Then you wouldn't need the arbitrary limit of 20 referendums and could benefit from future advances in the way the multi-index api will (hopefully) get improved. I know it would use more memory for the indices in the table but it feels more flexible. Am I missing a good reason against this?